connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/IncrementalCooperativeAssignor.java

```diff
@@ -753,6 +678,146 @@ private ConnectorsAndTasks assignment(Map<String, ConnectorsAndTasks> memberAssi
         ).build();
     }
 
+    /**
+     * Revoke connectors and tasks from each worker in the cluster until no worker is running more than it would be if:
+     * <ul>
+     *     <li>The allocation of connectors and tasks across the cluster were as balanced as possible (i.e., the difference in allocation size between any two workers is at most one)</li>
+     *     <li>Any workers that left the group within the scheduled rebalance delay permanently left the group</li>
+     *     <li>All currently-configured connectors and tasks were allocated (including instances that may be revoked in this round because they are duplicated across workers)</li>
```

Comment:
i couldn't follow this comment..

Reply:
When we detect duplicates, we revoke them from every worker that's running them, then wait for the next round to reassign them to a single worker. This part just clarifies that we don't take that into account when performing load-balancing revocations, and instead pretend that any duplicates that we're revoking in this round are allowed to be assigned. This is fine since the very next round should result in those duplicates being assigned, at which point a balanced allocation should be achieved.

Reply:
Does this make sense now? Any changes you'd like to see?